templates/reputation/contracts/ReputationTemplate.sol

```diff
@@ -0,0 +1,97 @@
+    uint64 constant private MIN_ACCEPTANCE_QUORUM = uint64(20 * ONE_PCT);   // 20%
+    uint64 constant private VOTE_DURATION = uint64(7 days);                 // 1 week
+    uint64 constant private FINANCE_PERIOD = uint64(30 days);               // 30 days
+
+    mapping (address => address) internal tokenCache;
+
+    constructor(DAOFactory _daoFactory, ENS _ens, MiniMeTokenFactory _miniMeFactory, IFIFSResolvingRegistrar _aragonID)
+        BaseTemplate(_daoFactory, _ens, _miniMeFactory, _aragonID)
+        public
+    {
+        // solium-disable-previous-line no-empty-blocks
+    }
+
+    function newTokenAndInstance(string id, address[] holders, uint256[] stakes) public {
+        require(holders.length == stakes.length, ERROR_INVALID_HOLDERS_STAKES_LEN);
+        newToken();
+        newInstance(id, holders, stakes);
+    }
+
+    function newToken() public returns (MiniMeToken) {
+        MiniMeToken token = _createToken(TOKEN_NAME, TOKEN_SYMBOL);
+        _cacheToken(token, msg.sender);
+        return token;
+    }
+
+    function newInstance(string id, address[] holders, uint256[] stakes) public {
+        require(holders.length == stakes.length, ERROR_INVALID_HOLDERS_STAKES_LEN);
+        MiniMeToken token = _popTokenCache(msg.sender);
+
+        // Create DAO and install apps
+        (Kernel dao, ACL acl) = _createDAO();
+        Agent agent = _installAgentApp(dao);
+        Finance finance = _installFinanceApp(dao, Vault(agent), FINANCE_PERIOD);
+        TokenManager tokenManager = _installTokenManagerApp(dao, token, TOKEN_TRANSFERABLE, TOKEN_MAX_PER_ACCOUNT);
+        Voting voting = _installVotingApp(dao, token, SUPPORT_REQUIRED, MIN_ACCEPTANCE_QUORUM, VOTE_DURATION);
+
+        // Mint tokens
+        _mintTokens(acl, tokenManager, holders, stakes);
+
+        // Set up permissions
+        _createAgentPermissions(acl, agent, voting, voting);
+        _createVaultPermissions(acl, Vault(agent), finance, voting);
+        _createFinancePermissions(acl, finance, voting, voting);
+        _createEvmScriptsRegistryPermissions(acl, voting, voting);
+        _createCustomVotingPermissions(acl, voting, tokenManager);
+        _createCustomTokenManagerPermissions(acl, tokenManager, voting);
+        _transferPermission(acl, voting, dao, dao.APP_MANAGER_ROLE());
+        _transferPermission(acl, voting, acl, acl.CREATE_PERMISSIONS_ROLE());
+
+        _registerID(id, dao);
+    }
+
+    function _mintTokens(ACL acl, TokenManager tokenManager, address[] holders, uint256[] stakes) internal {
+        _createPermission(acl, tokenManager, tokenManager.MINT_ROLE());
+        for (uint256 i = 0; i < holders.length; i++) {
+            tokenManager.mint(holders[i], stakes[i]);
+        }
+        _removePermission(acl, tokenManager, tokenManager.MINT_ROLE());
+    }
+
+    function _createCustomVotingPermissions(ACL acl, Voting voting, TokenManager tokenManager) internal {
```

Comment:
This is kind of the same but different than `_createVotingPermissions(ACL acl, Voting voting, TokenManager tokenManager)` in `DemocracyTemplate`?

Reply:
Democracy burns `MODIFY_SUPPORT_ROLE` permission, this one doesn't

Reply:
Oh, I see, but still, nitpick: `CompanyTemplate`, `DemocracyTemplate`, `MultisigTemplate` and `TrustTemplate` have `_createVotingPermissions`, while this one and `MembershipTemplate` have `_createCustomVotingPermissions`. I would unify the name (I lie it better without `Custom`, which is also more prevalent, but up to you, of course).
Besides, `TrustTemplate` has different parameters order: `(ACL acl, TokenManager tokenManager, Voting voting)`. I would use the same as the other ones.